src/Operations.js

```diff
@@ -389,6 +389,18 @@ export function groupByFactory(collection, grouper, context) {
   return groups.map(arr => reify(collection, coerce(arr))).asImmutable();
 }
 
+export function partitionFactory(collection, predicate, context) {
+  const isKeyedIter = isKeyed(collection);
+  const groups = [[], []];
```

Comment:
A Set has the wrong semantics because the result is ordered. As for using something else like a List, there are a couple of reasons why I didn't do it that way. One is that I expect the array to be immediately destructured in typical use, so creating an instance of any other type would add extra overhead that rarely has any benefit. The other is that it would break the TypeScript type of the result. In TypeScript the result is guaranteed to have exactly two values, and they even have different types when the predicate is a type-testing function.

I think the only other type that really makes sense is a native object type with fields named something like `true` and `false`; it would make the meaning the fields more clear, and it has no more overhead than an array. (I checked, and "true" and "false" work fine as field names.) Now that I think of it, I think returning an object is probably the the best, most idiomatic solution all around.

Reply:
Yeah, good point. I'm happy to leave it as-is.